app/assets/javascripts/discourse/templates/modal/edit-category.hbs

```diff
@@ -28,17 +28,19 @@
                  icon="trash-o"
                  label="category.delete"}}
     {{else}}
```

Comment:
You don't need all those changes in `edit-category.js.es6` and in this file if you replace the `{{else}}` with `{{else if model.id}}` 😉

Reply:
Ouch...that was so much better 😂
I've updated the code, thanks @gschlager!